domain/src/main/java/org/oppia/domain/audio/AudioPlayerController.kt

```diff
@@ -0,0 +1,112 @@
+package org.oppia.domain.audio
+
+import android.content.Context
+import android.media.MediaPlayer
+import android.net.Uri
+import android.provider.MediaStore
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import java.util.concurrent.Executors
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.locks.ReentrantLock
+import javax.inject.Inject
+import kotlin.concurrent.withLock
+
+/* AudioPlayerController controls interactions with an internal MediaPlayer,
+*  Use initializeMediaPlayer to set a source url for the audio
+*  Observe on getPlayState to get updates to status of MediaPlayer
+*  Use deinitializeMediaPlayer to release MediaPlayer and stop scheduling Seek Bar updates
+*/
+class AudioPlayerController @Inject constructor(val context: Context) {
+
+  inner class AudioMutableLiveData : MutableLiveData<PlayStatus>() {
+    override fun onActive() {
+      super.onActive()
+      seekBarActive = true
+    }
+
+    override fun onInactive() {
+      super.onInactive()
+      seekBarActive = false
+      stopUpdatingSeekBar()
+    }
+  }
+
+  enum class Status { PREPARED, COMPLETED, POSITION_UPDATE }
+  class PlayStatus(val type: Status, val position: Int, val duration: Int)
+
+  private val mediaPlayer: MediaPlayer by lazy { MediaPlayer() }
+  private val playState = AudioMutableLiveData()
+  private var nextUpdateJob: Job? = null
+  private val seekBarLock = ReentrantLock()
+
+  private var prepared = false
+  private var seekBarActive = false
+
+  private val SEEKBAR_UPDATE_FREQUENCY = TimeUnit.SECONDS.toMillis(1)
+
+  fun initializeMediaPlayer(url: String) {
```

Comment:
Here & elsewhere: any public function should have documentation comments associated with it to provide context on how to use the method, and errors that might occur.